sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala

```diff
@@ -783,7 +783,7 @@ class SQLContext(@transient val sparkContext: SparkContext)
 
   /** Returns the specified table as a [[DataFrame]]. */
   def table(tableName: String): DataFrame =
-    DataFrame(this, catalog.lookupRelation(Seq(tableName)))
+    DataFrame(this, UnresolvedRelation(Seq(tableName)))
```

Comment:
Thank you @yhuai for the explanation. I've created another PR (#4784), I'd like to return the `Unresolved` logical plan for `lookupRelation`, which will be helpful for the `describe extended cache|table|view|data source table`. Probably we'd better to review #4784 first and then jump back to this PR.